occurrence-download/src/main/java/org/gbif/occurrence/download/file/specieslist/SpeciesCount.java

```diff
@@ -0,0 +1,49 @@
+package org.gbif.occurrence.download.file.specieslist;
+
+import java.io.IOException;
+import java.util.Properties;
+import org.gbif.api.model.occurrence.Download;
+import org.gbif.api.service.registry.OccurrenceDownloadService;
+import org.gbif.occurrence.download.file.common.DownloadFileUtils;
+import org.gbif.occurrence.download.inject.DownloadWorkflowModule;
+import org.gbif.occurrence.download.util.RegistryClientUtil;
+import org.gbif.utils.file.properties.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.common.base.Preconditions;
+
+public class SpeciesCount {
```

Comment:
can you please be sure that we are not updating this field twice, the DownloadPrepareAction updates the total records, we should skip that in the DownloadPrepareAction for this type of downloads